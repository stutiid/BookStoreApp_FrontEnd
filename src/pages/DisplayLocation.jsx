import GoogleMapReact from "google-map-react";
import { Component } from "react";
import LocationOnRoundedIcon from "@mui/icons-material/LocationOnRounded";
import LocationDetails from "./LocationDetails";

/**
 * @description:- class is used to display the map and to spot the location selected by the user
 */
export default class DisplayLocation extends Component {
  static coordinates = {
    center: { lat: 20.5937, lng: 78.9629 },
    zoom: 6,
  };
  constructor(props) {
    super(props);
    this.state = {
      data: this.props.data,
    };
  }

  /**
   * @returns :- rendering of map on the page
   */
  render() {
    return (
      <div style={{ width: "100%", height: "1100px" }}>
        <GoogleMapReact
          className="display-map"
          defaultCenter={DisplayLocation.coordinates.center}
          defaultZoom={DisplayLocation.coordinates.zoom}
        >
          <LocationDetails
            icon={
              <LocationOnRoundedIcon
                style={{ fontSize: "large", color: "red" }}
              />
            }
            lat={this.state.data.latitude}
            lng={this.state.data.longitude}
            text={this.state.data.name}
            description={this.state.data.description}
            state={this.state.data.state}
            population={this.state.data.population}
          />
        </GoogleMapReact>
      </div>
    );
  }
}
