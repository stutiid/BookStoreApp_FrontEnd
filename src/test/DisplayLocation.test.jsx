import "./setupTest";
import DisplayLocation from "../pages/DisplayLocation";
import { shallow } from "enzyme";

const data = {
  id: 1,
  name: "Banglore",
  latitude: "12.9699",
  longitude: "77.5980",
  description:
    "Bengaluru (also called Bangalore) is the capital of India's southern Karnataka state.",
  state: "Karnataka",
  population: 13707000,
};

describe("tests", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<DisplayLocation data={data} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it("should render initial layout", () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it("has a component to render the map", () => {
    expect(wrapper.find(".display-map")).toBeTruthy();
  });
});
