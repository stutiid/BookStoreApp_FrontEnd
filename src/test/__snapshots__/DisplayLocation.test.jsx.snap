// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests should render initial layout 1`] = `
Array [
  <div
    style={
      Object {
        "height": "1100px",
        "width": "100%",
      }
    }
  >
    <t
      className="display-map"
      debounced={true}
      defaultCenter={
        Object {
          "lat": 20.5937,
          "lng": 78.9629,
        }
      }
      defaultZoom={6}
      distanceToMouse={[Function]}
      googleMapLoader={[Function]}
      heatmap={Object {}}
      heatmapLibrary={false}
      hoverDistance={30}
      layerTypes={Array []}
      options={[Function]}
      shouldUnregisterMapOnUnmount={true}
      style={
        Object {
          "height": "100%",
          "margin": 0,
          "padding": 0,
          "position": "relative",
          "width": "100%",
        }
      }
      yesIWantToUseGoogleMapApiInternals={false}
    >
      <LocationDetails
        description="Bengaluru (also called Bangalore) is the capital of India's southern Karnataka state."
        icon={
          <Memo
            style={
              Object {
                "color": "red",
                "fontSize": "large",
              }
            }
          />
        }
        lat="12.9699"
        lng="77.5980"
        population={13707000}
        state="Karnataka"
        text="Banglore"
      />
    </t>
  </div>,
]
`;
